### PR TITLE
doxygen updates for chgres_cube.fd/search_util.f90 and sfc_climo_gen.fd/interp.F90

### PR DESCRIPTION
**Pull request by lgannoaa**
This is the first try for the doxygen document effort. Welcome to all comments.

### CODE REVIEW

File: sorc/chgres_cube.fd/search_util.f90 [hunk 1]
```diff
@@ -1,12 +1,12 @@
 !> @file
-!! @brief Replace undefined surface values.
-!!
-!! @author gayno NCEP/EMC
-!!
-!! Replace undefined values with a valid value.  This can
-!! happen for an isolated lake or island that is unresolved by
-!! the input grid.
-!!
+!> @brief Replace undefined surface values.
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/search_util.f90:2]: Only the first comment in a doxygen block needs to be !>, the rest can be !!, so these need not be changed.

Intruducing blank lines changes the meening, so don't add the blank lines here.

Let's all use everyone's full name, so change ganyo to George Gayno everywhere it is used in a author tag, and the same for other names.

Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90]: I modified it again.

Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90]: Please take a look at this version, it contains update from a meeting took place today.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 2]
```diff
@@ -15,7 +15,7 @@ module search_util
 
  contains
 
-!> @brief Replace undefined surface values.
+!! @brief Replace undefined surface values.
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/search_util.f90:18]: Take out the brief tag for functions/subroutines. It's only needed for file-level documentation blocks.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/chgres_cube.fd/search_util.f90 [hunk 3]
```diff
@@ -16,24 +16,28 @@ module search_util
...
-!! does not consider valid values at adjacent faces.  That is a 
-!! future upgrade.
+!>
+!> Replace undefined values on the model grid with a valid value at
+!> a nearby neighbor.  Undefined values are typically associated
+!> with isolated islands where there is no source data.
+
+!> Routine searches a neighborhood with a radius of 100 grid points.
+!> If no valid value is found, a default value is used.
+
```
[the remark this answers is not in the document]
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90]: Blank line removed

File: sorc/chgres_cube.fd/search_util.f90 [hunk 4]
```diff
@@ -198,6 +202,10 @@ subroutine search (field, mask, idim, jdim, tile, field_num, latitude, terrain_l
 
  end subroutine search
 
+!> set sst values based on latitude
+!!
+!! @param latitude      - latitude input
+!! @param sst           - sst guess value to be set
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/search_util.f90:208]: Author tag please.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 5]
```diff
@@ -52,9 +56,8 @@ subroutine search (field, mask, idim, jdim, tile, field_num, latitude, terrain_l
  real(esmf_kind_r8)                :: field_save(idim,jdim)
  integer                           :: repl_nearby, repl_default
 
-!-----------------------------------------------------------------------
-! Set default value.
-!-----------------------------------------------------------------------
+!> @note Set default value.
```
[the remark this answers is not in the document]
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90]: This is outdated. I have modified it to remove note on last commit.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 6]
```diff
@@ -168,15 +167,15 @@ subroutine search (field, mask, idim, jdim, tile, field_num, latitude, terrain_l
...
+         !> @note Terrain heights for isolated landice points never get a correct value, so replace
+         !>  with terrain height from the input grid interpolated to the target grid
          field(i,j) = terrain_land(i,j)
          repl_default = repl_default + 1
        elseif (field_num == 224 .and. PRESENT(soilt_climo)) then
-          ! When using input soil type fields instead of climatological data on the
-          ! target grid, isolated land locations that exist in the target grid but
-          ! not the input grid don't receiving proper soil type information, so replace
-          ! with climatological values
+          !> @note When using input soil type fields instead of climatological data on the
```
[the remark this answers is not in the document]
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90]: This is outdated. I have modified it to remove note on last commit.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 7]
```diff
@@ -56,59 +56,59 @@ subroutine search (field, mask, idim, jdim, tile, field_num, latitude, terrain_l
...
  integer                           :: repl_nearby, repl_default
 
-!> @note Set default value.
-!> @return default value based on field_num
+!! @note Set default value.
+!! @return default value based on field_num
 
  select case (field_num)
-   case (0) ! most nst fields
+   case (0) !< most nst fields
```
[the remark this answers is not in the document]
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90:64]: Removed.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 8]
```diff
@@ -27,13 +27,17 @@ module search_util
 !! @note This routine works for one tile of a cubed sphere grid.  It
 !! does not consider valid values at adjacent faces.  That is a 
 !! future upgrade.
+!! @param terrain_land          - 2D field of terrain height points.
+!! @param soilt_climo           - 2D field of soil type points.
+
```
[the remark this answers is not in the document]
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90:32]: Removed

File: sorc/chgres_cube.fd/search_util.f90 (hunk 2 above)
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90:18]: Done

File: sorc/chgres_cube.fd/search_util.f90 (hunk 4 above)
Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90:208]: brief tag added

**Comment by edwardhartnett** [comment]
You still have converted the internal documentation to doxygen. Only the documentation blocks for functions/subroutines get converted to doxygen.

**Comment by lgannoaa** [comment]
I updated it again. Please check.

File: sorc/chgres_cube.fd/search_util.f90 (hunk 4 above)
Comment (edwardhartnett) [reply sorc/chgres_cube.fd/search_util.f90:205]: We do not need brief tag on subroutines/functions. We do need author tag.

Comment (lgannoaa) [reply sorc/chgres_cube.fd/search_util.f90:205]: Removed brief and add author tag

**Comment by lgannoaa** [comment]
Are we done with search_util.f90 at this point? Just want to confirm. Thanks

**Comment by edwardhartnett** [comment]
Yes, I think we are done. ;-)

**Comment by lgannoaa** [comment]
Now included the interp.F90 doxygen update. Please take a look.

File: sorc/sfc_climo_gen.fd/interp.F90 [hunk 9]
```diff
@@ -1,13 +1,12 @@
 !> @file
-!! @author gayno @date 2018
+!! @brief Read the input source data and interpolate it to the
+!! model grid.
 !!
-!! Read the input source data and interpolate it to the
-!! model grid. 
+!! @author gayno @date 2018
 !!
```
Comment (edwardhartnett) [inline sorc/sfc_climo_gen.fd/interp.F90:6]: In this case we do need a blank line here. There are two documentation blocks here, one for the file, and one for the subroutine interp

File: sorc/sfc_climo_gen.fd/interp.F90 [hunk 10]
```diff
@@ -259,30 +258,17 @@ subroutine interp(localpet, method, input_file)
...
- subroutine adjust_for_landice(field, vegt, idim, jdim, field_ch)
+!> Ensure consistent fields at land ice points
+!! Land ice is vegetation type 15 (variable landice).
+!! @author George Gayno NCEP/EMC
+!! Usage: call adjust_for_landice(field, vegt, idim, jdim, field_ch)
+!! output is Model field
+!! @param field         - Model field before adjustments for land ice.
+!! @param field_ch      - Field name.
+!! @param i/jdim        - i/j dimension of model tile.
+!! @param vegt          - Vegetation type on the model tile.
```
Comment (edwardhartnett) [inline sorc/sfc_climo_gen.fd/interp.F90:269]: author tag for subroutine as well as file.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/sfc_climo_gen.fd/interp.F90 (hunk 9 above)
Comment (lgannoaa) [reply sorc/sfc_climo_gen.fd/interp.F90:6]: Done

File: sorc/sfc_climo_gen.fd/interp.F90 (hunk 10 above)
Comment (lgannoaa) [reply sorc/sfc_climo_gen.fd/interp.F90:269]: Included.

**Comment by edwardhartnett** [comment]
@lgannoaa click the "Ready for review" if you are done with this PR and ready for it to be reviewed and merged. Right now it's a draft PR...

**Comment by lgannoaa** [comment]
"Ready for review" clicked. Thanks

**Comment by edwardhartnett** [comment]
You need to update it from the base branch. I'm seeing: This branch is out-of-date with the base branch. Normally it gives me a button to update the PR, but in this case there is no button.

When you submit the PR, if there is a place to select that repo admins can push changes to your fork, select that next time.

For now, just update the PR yourself, and I will merge it when done.

**Review by edwardhartnett — APPROVED** [review]

**Comment by lgannoaa** [comment]
I did a git fetch and merge those updates. It should be ready now.

**Pull request merged**